recipes/boost/all/conanfile.py

```diff
@@ -1330,18 +1046,10 @@ def add_defines(library):
 
         flags.append(f"toolset={self._toolset}")
 
-        safe_cppstd = self.settings.get_safe("compiler.cppstd")
-        if safe_cppstd:
-            cppstd_version = self._cppstd_flag(safe_cppstd)
-            flags.append(f"cxxstd={cppstd_version}")
-            if "gnu" in safe_cppstd:
-                flags.append("cxxstd-dialect=gnu")
-        elif Version(self.version) >= "1.85.0" and self._has_cppstd_14_supported:
-            cppstd_version = self._cppstd_flag("14")
-            flags.append(f"cxxstd={cppstd_version}")
-        elif self._has_cppstd_11_supported:
-            cppstd_version = self._cppstd_flag("11")
-            flags.append(f"cxxstd={cppstd_version}")
+        cppstd_version = cppstd_flag(self)
+        if "gnu" in cppstd_version:
+            flags.append("cxxstd-dialect=gnu")
+        flags.append(f'cxxstd={cppstd_version[-2:]}')
```

Comment:
It is not very clear this ``[-2:]`` range, what will happen if the returned flag is ``c++latest``?

Reply:
To clarify, boost ``cxxstd`` supports it:
> error: "None" is not a known value of feature <cxxstd>
error: legal values: "98" "03" "0x" "11" "1y" "14" "1z" "17" "2a" "20" "2b" "23" "2c" "26" "latest"

Reply:
The b2 uses cxxstd to define the standard, unlike cppstd in Conan: https://www.bfgroup.xyz/b2/manual/release/index.html#bbv2.overview.builtins.features

It also supports "latest", but indeed the current cut is wrong, as it will return "st" only. Thank you for spotting it.

Reply:
Fixed in the commit [ebb6b7a](https://github.com/conan-io/conan-center-index/pull/26030/commits/ebb6b7ad05d4e9a4154ab690efd302c59f1c24fa)